src/scrapers/leumi.js

```diff
@@ -119,6 +103,75 @@ async function fetchTransactionsForAccount(page, startDate) {
     }
   }
 
+  return txns;
+}
+
+async function extractPendingTransactionsFromPage(page) {
+  const txns = [];
+
+  const tdsValues = await page.$$eval('#WorkSpaceBox #trTodayActivityNapaTableUpper tr td', (tds) => {
+    return tds.map(td =>
+      ({
+        classList: td.getAttribute('class'),
+        innerText: td.innerText,
+      }));
+  });
+
+  for (const element of tdsValues) {
+    if (element.classList.includes('Colume1Width')) {
+      const newTransaction = { status: 'pending' };
+      newTransaction.date = (element.innerText || '').trim();
+      txns.push(newTransaction);
+    } else if (element.classList.includes('Colume2Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.description = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume3Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.reference = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume4Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.debit = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume5Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.credit = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume6Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.balance = element.innerText;
+      txns.push(changedTransaction);
+    }
```

Comment:
@esakal what's with all the pushes and pops?
Wouldn't it be much simpler to iterate over all `tr`s instead of all `td`s?
That way you will know when the row ends and could access a specific cell by index.

Reply:
@Urigo did it in his library at [line 206](https://github.com/Urigo/leumi-leumicard-bank-data-scraper/blob/master/src/scraping/leumi.js#L206). Since it works well and seems solid I think we should stick with what he did.

Reply:
@esakal 😞still don't like it, but if you insist...